### DISCUSSION TITLE
analog pot can't reach full value when turn fast

### DISCUSSION THREAD

**Discussion by ratracerecord** [shanteacontrols/OpenDeck#171] [Q&A] [answered]
Hello back!

Now i got a board working i got an issue with analog pot (https://www.mouser.fr/ProductDetail/652-91R1A-R16-A15L) on opendeck v3.0.1 pcb.

EDIT : i forgot to say i'm using the opendeck v3 board with arduino nano 33 ble.

when i turn the pot fast the midi don't reach the full value.
when i turn it slow it goes nicely to the full value.

i made a video here ( https://www.youtube.com/shorts/UO9lGMR6wWk )

Best regards,

Jules.

**Comment by paradajz** [comment]
I don't understand which board are you using - bare Nano 33 BLE or the OpenDeck v3?

**Reply by ratracerecord** [reply]
opendeck v3 (the A version with nano 33 ble on it)

**Reply by paradajz** [reply]
Okay, and which firmware do you have loaded on it?

**Reply by ratracerecord** [reply]
V 7.3.2


![Sans titre](https://user-images.githubusercontent.com/111751040/186418711-70bb6ba0-3a4f-4cdc-86cc-2733eb459232.png)

**Reply by paradajz** [reply]
Okay. Can you post a picture of your wiring and screenshot of analog section in the configuration with the view set to Table?

**Comment by ratracerecord** [comment]
![Sans titre2](https://user-images.githubusercontent.com/111751040/186419641-0796c87a-bd93-4dd5-9762-f1181449d9bf.png)
![IMG_20220824_143253](https://user-images.githubusercontent.com/111751040/186419650-37a25e1e-ecb5-4fef-a2fe-7db22ec037f7.jpg)

**Comment by paradajz** [comment, marked as answer]
The upper limit offset is set to 10. Intentionally?

**Reply by ratracerecord** [reply]
i was testing if it help but it is the same at 0

**Reply by paradajz** [reply]
[opendeck3a.zip](https://github.com/shanteacontrols/OpenDeck/files/9416429/opendeck3a.zip)

Try flashing the attached firmware. Seems like I've set sample time a bit too high. Let me know if this introduces some instability to analog readings.

**Reply by ratracerecord** [reply]
it should show as 7.3.2 too ?

if yes after firmware update it seem not chaging a lot (maybe a bit better the lowest value when i turn fast is now in the range of 60 - 80 rather than 20 - 60 before).

**Reply by paradajz** [reply]
Yes the version is unchanged. That's weird since I cannot reproduce it anymore - no matter how fast I move it it always reaches 0/127.

**Reply by paradajz** [reply]
But here's another reduction anyways.

[opendeck3a.zip](https://github.com/shanteacontrols/OpenDeck/files/9416820/opendeck3a.zip)

**Reply by ratracerecord** [reply]
donst seem to change anything.

But i found other issue with analog pot.

i was triying another pot (bourn 10k conductive plastique rather than cemet for the first tested) and when i connect booth and try to use booth something weird happened : 
when booth pot are at 0 i can use one or the other without probleme BUT when one of the two pot is at 127 the other one can't work at all (turning the pot don't entry new value).

**Reply by paradajz** [reply]
Then something is wrong at your end. I cannot reproduce such issues here.

**Reply by paradajz** [reply]
You also need to short-circuit the VUSB pad on the bottom on the Nano 33 BLE.

**Reply by ratracerecord** [reply]
the VUSB is done, i got 5V on the board.

**Reply by ratracerecord** [reply]
i'm gona try the nano hex file without the opendeck board

**Reply by ratracerecord** [reply]
works perfect with pot connect directly tothe nano pins with the nano hex, no miss full value.

**Reply by ratracerecord** [reply]
without knoing how it works now, for the full value and for the use off two pot at the same time...

**Reply by ratracerecord** [reply]
i flash the last opendeck firmware you send rather than uploading it via bootloder mode via the web interface.

**Comment by ratracerecord** [comment]
For your information @paradajz , i got the same issues (random value at full  when turn fast, and no reaction of one pot when the other is at full value) with the last firmware update (7.3.3), so i go back on the last you gave me here who's working fine.

> But here's another reduction anyways.
> 
> [opendeck3a.zip](https://github.com/shanteacontrols/OpenDeck/files/9416820/opendeck3a.zip)

And btw, i wasnt able to upload the new firmware with web interface and should had go to the DK and flash the nano.

Best regards!

Jules.

**Reply by paradajz** [reply]
Yes, hex files need to be flashed with programmer. Anyways, I am unable to reproduce the issue on my boards here with the latest release, so there's that.

**Reply by ratracerecord** [reply]
Ok but the last hex you made in this topic made it work for me.
i don't know what you change in this one.

Thank you for your time!

**Reply by paradajz** [reply]
> Ok but the last hex you made in this topic made it work for me

You said this about it yesterday:

> donst seem to change anything.

In the last hex I've changed sampling time to almost a minimum which started to affect stability on my test setup (2 samples per reading). v7.3.3 uses 8 samples, and everything before that used 16 samples.

**Reply by ratracerecord** [reply]
sorry we miss understoud. the first time i failed uploading the hex via web interface (i was thinking it could work AND had work, so i don't see modification).

But when i try flashing with programmer it warked nice with it.

in my situation 2 samples seems fine in my setup.

**Reply by paradajz** [reply]
Glad it works out, although I think it's not an issue with sample rate at all anymore since yesterday, at some point, I was able to reproduce the problem you had with the original firmware (v7.3.2) and then later I couldn't anymore. I think it has something to do with the ADC calibration phase on nRF52 but I'm not sure. I've reduced the sample rate to 8 for the latest release anyways since turning the pot real fast seemed to cause slight but noticeable delay. Reduction fixed that but I am unable to reproduce the problem you had anymore - either with v7.3.2 or the v7.3.3. I'll update the thread if I discover something new.

**Reply by ratracerecord** [reply]
thank you for explanation !
